### DISCUSSION TITLE
Subsequent geom_tile call is not zero indexed for x axis.

### DISCUSSION THREAD

**Discussion by joseph-jnl** [has2k1/plotnine#905] [Q&A] [answered]
Hello, I'm trying to understand why my second call to `geom_tile` is not zero indexed like the first.

![image](https://github.com/user-attachments/assets/a8d7faf2-39d8-4a13-98ba-3103eb41336e)

i.e.. The first `geom_tile` uses zero indexed rows and columns but the second `geom_tile` call starts at 1 instead for the x axis. The intent is to highlight the tiles at row 0 and column 1 and 3 rather than columns 0 and 2.

Thank you!

**Comment by has2k1** [comment]
Please post a minimal reproducible example.

**Comment by joseph-jnl** [comment]
```python
import pandas as pd
from plotnine import ggplot, aes, geom_tile, scale_y_reverse

df1 = pd.DataFrame([[0,1,2], [3,4,5], [6,7,8]])
df1 = df1.reset_index().melt('index')
df1.columns = ['row', 'column', 'value']
df2 = pd.DataFrame({"row": [0, 0], "column": [1, 2], "value": list(range(2))})

(ggplot(df1, aes("column", "row", fill="value")) 
 + geom_tile()
 + geom_tile(data=df2, 
             size=2, fill=None, color="black")
 + scale_y_reverse()
 )
```

![image](https://github.com/user-attachments/assets/15de4ac9-b420-46de-8d8c-81e7571b8756)

The current output aligns with what I would expect a `df2` with `column` set to [0, 1].

**Reply by has2k1** [reply, marked as answer]
With the way you create the dataframe, `df1["column"]` is discrete (dtype=object) while `df2["column"]` is continuous (dtype=int).

```python
print({k: str(v.dtype) for k, v in df1.items()})
print({k: str(v.dtype) for k, v in df2.items()})

# {'row': 'int64', 'column': 'object', 'value': 'int64'}
# {'row': 'int64', 'column': 'int64', 'value': 'int64'}
```

So if you do

```python
df1["column"] = df1["column"].astype(int)
```

It should act as you expect.

**Comment by joseph-jnl** [comment]
Just to close the loop- posting the final plots. The expressiveness of plotnine is second to none (e.g. Creating policy grid arrows using layered spoke geoms), thanks again!  

![image](https://github.com/user-attachments/assets/4557c45f-302b-4442-aeb9-b2bd3fa8ccb6)

![image](https://github.com/user-attachments/assets/dd3a6064-abf4-416c-9a18-a1ccd5277c3e)

https://joseph-jnl.github.io/rlbook/chap3_finite_mdps/
